src/services/textChanges.ts

```diff
@@ -451,33 +466,45 @@ namespace ts.textChanges {
                 return "";
             }
             const options = change.options || {};
-            const nonFormattedText = getNonformattedText(change.node, sourceFile, this.newLine);
+            let text: string;
+            const pos = change.range.pos;
+            const posStartsLine = getLineStartPositionForPosition(pos, sourceFile) === pos;
+            if (isArray(change.node)) {
+                const parts = change.node.map(n => this.getFormattedTextOfNode(n, sourceFile, pos, change.useIndentationFromFile, options));
+                text = parts.join(change.options.nodesSeparator);
+            }
+            else {
+                text = this.getFormattedTextOfNode(change.node, sourceFile, pos, change.useIndentationFromFile, options);
+            }
+            // strip initial indentation (spaces or tabs) if text will be inserted in the middle of the line
+            // however keep indentation if it is was forced
+            text = posStartsLine || change.options.indentation !== undefined ? text : text.replace(/^\s+/, "");

```

Comment:
We don't have a leading trim function yet; since we'd theoretically have to scan for all the Unicode whitespace I'm not sure it can be done more cheaply than that RegEx.

Added parens for clarity (the `||` expression is the first operand to `?`)